accounts/src/inmemory_account_store.rs

```diff
@@ -240,12 +244,11 @@ impl InmemoryAccountStore {
 
```

Comment:
`update_owner` really looks strange.

this looks like an assertion: `prev_account_data.pubkey == new_account_data.pubkey`

this should fast-return `prev_account_data.account.owner == new_account_data.account.owner`

the finalized case could call a method `self.set_only_account_owner` which is a variant of `self.add_account_owner`